src/main/java/net/minestom/server/item/drop/DropReason.java

```diff
@@ -0,0 +1,17 @@
+package net.minestom.server.item.drop;
+
+import net.minestom.server.inventory.AbstractInventory;
+import org.jetbrains.annotations.NotNull;
+
+public class DropReason {
```

Comment:
Nitpick: Make it final for hardening api or make it into sealed interface

Reply:
Add also javadocs.

Reply:
The entire point of having the DropReason class is that server implementations can make their own version if they need it (say if they create another reason to drop an item, like DropReasonHitByMob).
Making that an interface and sealed would defeat the point of it.

We won't have generated data for this.

I've addressed your other nitpicks.

Reply:
The point is that we should rather do this with events and an interface structure but that should be hardened by the server itself. 

I have the following in mind:
```java
// This class is harded
public sealed interface DropReason permits CustomDropReason {
}
// This class can be extend from User
public non-sealed interface CustomDropReason extends DropReason {
}
```